src/containers/pages/PractitionerViews/helpers/serviceHooks.ts

```diff
@@ -40,3 +41,45 @@ export const loadPractitioner = async (
       growl(err.message, { type: toast.TYPE.ERROR });
     });
 };
+
+/**
+ * load all practioners with pagination
+ * @param { typeof OpenSRPService } serviceClass -  the OpenSRP service
+ * @param { number } pageSize - number of records per page
+ * @param { number } pageNumber - page to fetch records from
+ * @param { boolean } getAll - determine if to get all availble pages recusively
+ */
+export const getAllPractitioners = (
+  serviceClass: typeof OpenSRPService,
+  pageSize: number = 1000,
+  pageNumber: number = 1,
+  getAll: boolean = false
+) => {
+  const service = new serviceClass(OPENSRP_PRACTITIONER_ENDPOINT);
+
+  const getPractitionersByPage = (currentPage: number, results: Practitioner[] = []) => {
+    const filterParamss = {
+      pageNumber: currentPage,
+      pageSize,
+    };
+    if (results.length || pageNumber === currentPage) {
+      return service
+        .list(filterParamss)
+        .then((res: Practitioner[]) => {
+          if (res && res.length) {
+            store.dispatch(fetchPractitioners(res));
+          }
+          if (getAll) {
+            const newPage = ++currentPage;
+            // tslint:disable-next-line: no-floating-promises
+            getPractitionersByPage(newPage, res);
```

Comment:
What will happen if an exception occurs here?

Reply:
will through an error.

Reply:
will the error be caught nicely?

Reply:
Yeah, this [line](https://github.com/onaio/reveal-frontend/blob/54d7912fad75c2d88f897b4fec01d8336941941e/src/containers/pages/PractitionerViews/helpers/serviceHooks.ts#L78) catches any exceptions.